sys/suit/v4/handlers.c

```diff
@@ -0,0 +1,548 @@
+
+        const riotboot_hdr_t *hdr = riotboot_slot_get_hdr(riotboot_slot_current());
+        if (seq_nr <= (int64_t)hdr->version) {
+            LOG_INFO("%"PRIu64" <= %"PRIu32"\n", seq_nr, hdr->version);
+            LOG_INFO("seq_nr <= running image\n)");
+            return -1;
+        }
+
+        hdr = riotboot_slot_get_hdr(riotboot_slot_other());
+        if (riotboot_hdr_validate(hdr) == 0) {
+            if (seq_nr <= (int64_t)hdr->version) {
+                LOG_INFO("%"PRIu64" <= %"PRIu32"\n", seq_nr, hdr->version);
+                LOG_INFO("seq_nr <= other image\n)");
+                return -1;
+            }
+        }
+
+        LOG_INFO("suit: validated sequence number\n)");
+        manifest->validated |= SUIT_VALIDATED_SEQ_NR;
+        return 0;
+    }
+    LOG_INFO("Unable to get sequence number\n");
+    return -1;
+}
+
+static int _dependencies_handler(suit_v4_manifest_t *manifest, int key,
+                                 CborValue *it)
+{
+    (void)manifest;
+    (void)key;
+    (void)it;
+    /* No dependency support */
+    return 0;
+}
+
+static int _component_handler(suit_v4_manifest_t *manifest, int key,
+                              CborValue *it)
+{
+    (void)manifest;
+    (void)key;
+
+    CborValue arr;
+
+    LOG_DEBUG("storing components\n)");
+    if (!cbor_value_is_array(it)) {
+        LOG_DEBUG("components field not an array\n");
+        return -1;
+    }
+    cbor_value_enter_container(it, &arr);
+
+    unsigned n = 0;
+    while (!cbor_value_at_end(&arr)) {
+        CborValue map, key, value;
+        if (n < SUIT_V4_COMPONENT_MAX) {
+            manifest->components_len += 1;
+        }
+        else {
+            LOG_DEBUG("too many components\n)");
+            return SUIT_ERR_INVALID_MANIFEST;
+        }
+
+        suit_cbor_map_iterate_init(&arr, &map);
+
+        suit_v4_component_t *current = &manifest->components[n];
+
+        while (suit_cbor_map_iterate(&map, &key, &value)) {
+            // handle key, value
+            int integer_key;
+            if (suit_cbor_get_int(&key, &integer_key)) {
+                return SUIT_ERR_INVALID_MANIFEST;
+            }
+
+            switch (integer_key) {
+                case SUIT_COMPONENT_IDENTIFIER:
+                    current->identifier = value;
+                    break;
+                case SUIT_COMPONENT_SIZE:
+                    LOG_DEBUG("skipping SUIT_COMPONENT_SIZE");
+                    //current->size = value;
+                    break;
+                case SUIT_COMPONENT_DIGEST:
+                    current->digest = value;
+                    break;
+                default:
+                    LOG_DEBUG("ignoring unexpected component data (nr. %i)\n", integer_key);
+            }
+
+            LOG_DEBUG("component %u parsed\n", n);
+        }
+
+        cbor_value_advance(&arr);
+        n++;
+    }
+
+    manifest->state |= SUIT_MANIFEST_HAVE_COMPONENTS;
+    cbor_value_enter_container(it, &arr);
```

Comment:
This isnt doing anything here, shouldn't it be `cbor_value_leave_container(...)`